lib/puppet/file_serving/terminus_helper.rb

```diff
@@ -12,6 +12,9 @@ def path2instance(request, path, options = {})
     result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
     result.source_permissions = request.options[:source_permissions] if request.options[:source_permissions]
 
+    result.environment = request.environment
```

Comment:
Gating this environment setter on the `static_catalog` option would avoid a bunch of the changes to tests below, but I kind of am happier with the terminus_helper not knowing the details of the models (specifically, that environment is only used when statically compiling)